### PR TITLE
New boilerplate parser confidence refactor

### PR DESCRIPTION
**Pull request by greglittlefield-wf**
### Major Changes:
- Be more strict 
- Move version confidence logic out of member classes and into member detection. This:
    - allows for computation before we create member instances, so we don't have to filter them out after the fact.
    - unifies the "is this a boilerplate member" logic with the "which version is it" logic, since the two have a lot of overlap and resulted in duplicate code
- Remove `BoilerplateVersion.noGenerate`; instead, use a confidence threshold to determine whether something should be generated. For declarations, this is the mean confidence of all members.
    - This allows us to easily determine whether leftover boilerplate members should result in warning/errors (above threshold) or be ignored (below threshold)
    - This removes the multiple ways to say something shouldn't be generated (low confidence in boilerplates vs high confidence in `noGenerate`)

- Create classes for version confidence instead of using a Map, which was more susceptible to missing keys

### Minor changes / fixes
- Find components' props by the UiProps parameter if the name doesn't match
- Rename `BoilerplateVersion` -> `Version` since it was too long lol
- Fix inverted `classish.abstractKeyword` null-check
- Clean up `BoilerplateMemberDetector` / decouple it from BoilerplateMembers

### CODE REVIEW

**Comment by aviary3-wk** [comment]
## Security Insights
No security relevant content was detected by automated scans.

## Action Items
* Review PR for [security impact](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); comment "security review required" if needed or unsure
* Verify [`aviary.yaml`](https://dev.workiva.net/docs/teams/information-security/aviary#configuring-aviary) coverage of security relevant code
---
_Questions or Comments? Reach out on Slack: #support-infosec._

File: lib/src/builder/generation/parsing/declarations.dart [hunk 1]
```diff
@@ -49,14 +49,6 @@ Iterable<BoilerplateDeclaration> getBoilerplateDeclarations(
   final states = members.states;
   final stateMixins = members.stateMixins;
 
-  // -----------------------------------------------------------------------------------------------
-  //
-  // Legacy map views
-  //
-  // Remove them so they don't get grouped with anything else and throw things off.
-  //
-  props.removeWhere((p) => p.isLegacyMapView);
```
Comment (greglittlefield-wf) [inline lib/src/builder/generation/parsing/declarations.dart]: I can't remember why this isn't needed anymore... 😬 I might want to put it back?

File: lib/src/builder/generation/parsing/version.dart [hunk 2]
```diff
@@ -5,58 +5,114 @@ import 'members.dart';
...
+    final confidenceMap = {
+      'v2_legacyBackwardsCompat': v2_legacyBackwardsCompat,
+      'v3_legacyDart2Only': v3_legacyDart2Only,
+      'v4_mixinBased': v4_mixinBased,
+    };
+
+    return '${runtimeType.toString()}; confidence:$confidenceMap';
+  }
+}
+
```
Comment (joebingham-wk) [inline lib/src/builder/generation/parsing/version.dart]: #nit extra line
```suggestion
```

File: lib/src/builder/generation/parsing/declarations.dart (hunk 1 above)
Comment (greglittlefield-wf) [reply lib/src/builder/generation/parsing/declarations.dart]: Good call, okay, I'll reinstate it and add a todo to figure out if we can remove it

**Pull request merged**